### PR TITLE
Feat/dv 54/create main page layout

### PR DESCRIPTION
**Pull request by vadymchornyidev**

### CODE REVIEW

File: src/assets/scss/assets/components/_header.scss [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            margin: 0%;
+        }
+        #change-theme img{
+            width: 0px;
+        }
+        .support_block{
+            width: 25%;
+        }
+    }
+}
```
Comment (marynakernychna) [inline src/assets/scss/assets/components/_header.scss]: Empty line at EoF

File: src/assets/scss/assets/components/_menu.scss [hunk 2]
```diff
@@ -0,0 +1,167 @@
+@import '../variables/colors';
+@import '../variables/others';
+
+.Menu {
```
Comment (marynakernychna) [inline src/assets/scss/assets/components/_menu.scss]: we call classes in lowercase

Comment (marynakernychna) [reply src/assets/scss/assets/components/_menu.scss]: why don't you use the nesting?

File: src/assets/scss/assets/components/_header.scss (hunk 1 above)
Comment (vladtymo) [inline src/assets/scss/assets/components/_header.scss]: Add empty line here and check the other files

File: src/assets/scss/assets/components/_menu.scss [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+     }
+     .Menu-items{
+         display: flex;
+         align-items: center;
+     }
+     .logout{
+         margin-top: 25px;
+     }
+    }
+}
```
Comment (marynakernychna) [inline src/assets/scss/assets/components/_menu.scss]: Empty line at EoF

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 4]
```diff
@@ -2,18 +2,70 @@
...
+    .role-block{
+        display: flex;
+        justify-content: center;
+        align-items: center;
+    }
+    
+    .block-driver{
+        width: 250px;
+    }
+}
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss]: Empty line at EoF

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 5]
```diff
@@ -2,18 +2,70 @@
 
 .mainPageBody {
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss:6]: why don't you use the nesting?

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 6]
```diff
@@ -2,18 +2,70 @@
...
+    height: 250px;
+    border-radius: 10px;
+    color: #fff;
+    font-size: 20px;
+    background-size: cover;
+
+}
+
+.block-sender{
+    background-image: url('https://kufer.media/wp-content/uploads/2020/07/72239050_2483513021766295_8043596575647203328_o.jpg');
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss]: what will happen if the resource is removed from the site?

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 7]
```diff
@@ -2,18 +2,70 @@
...
 }
+
+.block-recipiante{
+    background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTj8cOHNbNl6SmIzIG-kvw9h-0QKj_dyTO57KeAFSRDiPRYrCyNfKAaY74MNXUk4CzWCGo&usqp=CAU');
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss]: what will happen if the resource is removed from the site?

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 8]
```diff
@@ -2,18 +2,70 @@
...
+
+}
+
+.block-driver{
+    background-image: url('https://www.betterteam.com/images/driver-job-description-5558x3706-2020123.jpeg?crop=40:21,smart&width=1200&dpr=2');
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss]: what will happen if the resource is removed from the site?

File: src/assets/scss/assets/pages/_mainPage.scss (hunk 6 above)
Comment (vladtymo) [inline src/assets/scss/assets/pages/_mainPage.scss]: I think better to load the resources and use them locally

File: src/assets/scss/assets/variables/_colors.scss [hunk 9]
```diff
@@ -3,3 +3,9 @@ $body_color: #093545;
 $input_color: #224957;
 $header_color: #05192c;
 $input_error_message_color: rgb(211, 19, 19);
+
+$main_header_color: #FAFAFA;
+$white_gray: #DDFBE7;
+$white_light_gray: #FFF6F6;
+$white: #fff;
+$black: black;
```
Comment (marynakernychna) [inline src/assets/scss/assets/variables/_colors.scss]: what is it for? By default, the color has this name

File: src/assets/scss/assets/variables/_others.scss [hunk 10]
```diff
@@ -5,3 +5,8 @@ sans-serif;
 
 $max_phone_size: 600px;
 $max_tablet_size: 1200px;
+$min_max_monitor_size: 1745px;
+
```
Comment (vladtymo) [inline src/assets/scss/assets/variables/_others.scss:10]: Remove the unnecessary spaces

File: src/assets/scss/main.scss [hunk 11]
```diff
@@ -1,2 +1,4 @@
 @import "./assets/pages/authentication";
 @import "./assets/pages/mainPage";
+@import "./assets/components/header";
+@import "./assets/components/menu"
```
Comment (marynakernychna) [inline src/assets/scss/main.scss]: Empty line at EoF

File: src/assets/scss/assets/variables/_others.scss [hunk 12]
```diff
@@ -5,3 +5,8 @@ sans-serif;
...
+
```
Comment (marynakernychna) [inline src/assets/scss/assets/variables/_others.scss]: extra empty line

File: src/components/header.jsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        else setLanguage(ua);
+    };
+
+    const changeTheme = () => {
+        if (theme === whiteTheme) setTheme(darkTheme);
+        else setTheme(whiteTheme);
+    };
+
+    const [isOpen, setIsOpen] = useState(false);
+
```
Comment (marynakernychna) [inline src/components/header.jsx]: extra empty line

File: src/components/header.jsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        if (language === ua) setLanguage(en);
+        else setLanguage(ua);
+    };
+
+    const changeTheme = () => {
+        if (theme === whiteTheme) setTheme(darkTheme);
+        else setTheme(whiteTheme);
+    };
+
+    const [isOpen, setIsOpen] = useState(false);
```
Comment (marynakernychna) [inline src/components/header.jsx]: why are these fields here and not together with others?

File: src/components/menu.jsx [hunk 15]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from "react";
+import { items } from "./menuItems";
+import { logoutUser } from "../services/authentication";
+import { useHistory } from "react-router-dom";
+import "../assets/scss/assets/components/_menu.scss";
```
Comment (marynakernychna) [inline src/components/menu.jsx]: scss must not import anywhere other than src/index.js

File: package.json [hunk 16]
```diff
@@ -9,8 +9,10 @@
     "@testing-library/user-event": "^13.5.0",
     "antd": "^4.19.5",
     "axios": "^0.26.1",
+    "jwt-decode": "^3.1.2",
```
Comment (marynakernychna) [inline package.json]: not used anywhere

File: src/assets/scss/assets/components/_menu.scss [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+    }
+    100% {
+        opacity: 1;
+        transform: translateY(0);
+    }
+}
+
+// Big monitor
+
+@media only screen and (min-width: 1600px) and (max-width: $min_max_monitor_size) {
```
Comment (marynakernychna) [inline src/assets/scss/assets/components/_menu.scss]: why are all dimensions constant except this?

File: src/assets/scss/assets/variables/_others.scss [hunk 18]
```diff
@@ -5,3 +5,8 @@ sans-serif;
 
 $max_phone_size: 600px;
 $max_tablet_size: 1200px;
+$min_max_monitor_size: 1745px;
```
Comment (marynakernychna) [inline src/assets/scss/assets/variables/_others.scss]: so is the minimum or maximum size?

File: src/assets/scss/main.scss [hunk 19]
```diff
@@ -1,2 +1,4 @@
 @import "./assets/pages/authentication";
 @import "./assets/pages/mainPage";
+@import "./assets/components/header";
```
Comment (marynakernychna) [inline src/assets/scss/main.scss]: it should not be here

File: src/assets/scss/main.scss (hunk 11 above)
Comment (marynakernychna) [inline src/assets/scss/main.scss]: it should not be here

File: src/components/header.jsx [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+
+    const [data, setData] = useState("Null");
+
+    useEffect(() => {
+        userService.getUser().then((res) => {
```
Comment (marynakernychna) [inline src/components/header.jsx]: I think it should be taken to the service, not called here

File: src/components/mainPage/index.jsx [hunk 21]
```diff
@@ -1,24 +1,20 @@
-import React from "react";
-import { useHistory } from "react-router-dom";
-import { ExportOutlined } from '@ant-design/icons';
-import { logoutUser } from "../../services/authentication";
+import React, { useEffect, useState } from "react";
```
Comment (marynakernychna) [inline src/components/mainPage/index.jsx]: where do you use it (useEffect, useState) here?

File: src/components/mainPage/index.jsx [hunk 22]
```diff
@@ -1,24 +1,20 @@
...
+import Header from "../header";
+import { userService } from "../../api/user";
```
Comment (marynakernychna) [inline src/components/mainPage/index.jsx]: where do you use it here?

File: src/components/menu.jsx [hunk 23]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from "react";
```
Comment (marynakernychna) [inline src/components/menu.jsx]: where do you use it (useEffect, useState) here?

**Review by marynakernychna — COMMENTED** [review]
You need to create a folder for everything (menu.jsx & manuItems.js) related to the menu

File: src/components/menuItems.js [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+        itemIcon:
+            "https://cdn-icons.flaticon.com/png/512/2211/premium/2211642.png?token=exp=1651566256~hmac=2aa22f2509cf5c41bb5c15f81f6e025f",
+        itemText: "Change role",
+        itemLink: "/main",
+    },
+    {
+        itemIcon:
+            "https://cdn-icons.flaticon.com/png/512/2962/premium/2962303.png?token=exp=1651566274~hmac=daeef7d8eeb18ec6e915be3c1e058dfd",
+        itemText: "My cars",
+        itemLink: "/myCars",
```
Comment (marynakernychna) [inline src/components/menuItems.js]: maybe we should use the route format like "/my-cars" or "/cars"

Comment (marynakernychna) [reply src/components/menuItems.js]: and for other site routes that are in this file

**Review by marynakernychna — COMMENTED** [review]
Screenshot 2022-05-04 134739

Why the menu is so huge? Where is the dark theme icon & the menu icons?

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 25]
```diff
@@ -2,18 +2,70 @@
...
     }
+}
 
-    div {
-        display: inline-block;
-        width: 200px;
-        color: white;
-        margin: 5px 20px;
-    }
+.role-container{
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss:15]: I advise you to darken the photos, the text is poorly visible

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 26]
```diff
@@ -1,19 +1,74 @@
...
+
 .mainPageBody {
     height: 100vh;
-    background-color: $body_color;
-
-    header {
-        background-color: $header_color;
-        height: 54px;
-        text-align: end;
+    background-color: $white;
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss]: is it some unusual white?  why is it written as a variable?

File: src/components/navigation/header.jsx [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+        else setLanguage(ukraine_language);
+    };
+
+    const changeTheme = () => {
+        if (theme === white_theme) setTheme(dark_theme);
+        else setTheme(white_theme);
+    };
+
+    useEffect(() => {
+        userService.getUser().then((res) => {
```
Comment (marynakernychna) [inline src/components/navigation/header.jsx]: I think it should be taken to the service, not called here. And be called when the main page is loaded

File: src/assets/scss/assets/components/_menu.scss [hunk 28]
```diff
@@ -0,0 +1,167 @@
+@import '../variables/colors';
+@import '../variables/others';
+
+.menu {
```
Comment (marynakernychna) [inline src/assets/scss/assets/components/_menu.scss:4]: Icons in the menu are not displayed
Screenshot 2022-05-04 180327

File: src/services/userService.js [hunk 29]
```diff
@@ -0,0 +1,15 @@
+import { useContext } from "react";
+import userService from "../api/user";
+import { UserContext } from "../components/context/user.context";
+
+export function GetUserName() {
```
Comment (marynakernychna) [inline src/services/userService.js]: we call functions in lowercase

File: src/components/mainPage/index.jsx [hunk 30]
```diff
@@ -1,24 +1,20 @@
...
+import { GetUserName } from "../../services/userService";
 
 function MainPage() {
-
-    let history = useHistory();
-
-    const logOut = () => {
-        logoutUser(history);
-    };
+    GetUserName();
```
Comment (marynakernychna) [inline src/components/mainPage/index.jsx]: Why it hangs in the air and is not called at the start of the page ? It is called three times.

Screenshot 2022-05-04 211802

File: src/assets/scss/assets/pages/_mainPage.scss [hunk 31]
```diff
@@ -1,19 +1,74 @@
...
+
+.role-container{
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+    margin-top: 3%;
+}
+
+.role-block{
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_mainPage.scss:23]: Role names, such as Recipient, are difficult to see

Screenshot 2022-05-04 213727

File: src/services/userService.js [hunk 32]
```diff
@@ -0,0 +1,18 @@
+import userService from "../api/user";
+
+export function getUserName() {
+    return userService
+        .getUser()
+        .then(
+            (response) => {
+                console.log(response.data.name + " " + response.data.surname);
```
Comment (marynakernychna) [inline src/services/userService.js]: console.log() musn't be here and you need to display messages to the user

**Review by marynakernychna — APPROVED** [review]

**Review by vladtymo — APPROVED** [review]

**Pull request merged**